.github/workflows/e2e-tests.yaml

```diff
@@ -0,0 +1,72 @@
+name: End-to-end tests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```

Comment:
Since this takes ~12m to run, we can keep it on `master` and for daily runs. We could also keep it as a non-required workflow for PRs. Let me know what you think.